File: Module4/Casestudy/case_study/case_study/src/main/java/com/codegym/casestudy/service/employee/EmployeeService.java
package com.codegym.casestudy.service.employee;

import com.codegym.casestudy.model.employee.Employee;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface EmployeeService {
    Page<Employee> findAll(Pageable pageable);

    void save(Employee employee);

    void delete(Long id);

    Employee findById(Long id);
}

File: Module5/case_study2/web3/src/app/service/employee.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Employee} from '../model/employee';
const API_URL = 'http://localhost:3000';

@Injectable({
  providedIn: 'root'
})
export class EmployeeService {

  constructor(private httpClient: HttpClient) { }

  getAll(): Observable<Employee[]> {
    return this.httpClient.get<Employee[]>(API_URL + '/employee');
  }

  save(employee: Employee): Observable<Employee> {
    return this.httpClient.post<Employee>(API_URL + '/employee', employee);
  }

  findById(id: number): Observable<Employee> {
    return this.httpClient.get<Employee>(`${API_URL}/employee/${id}`);
  }

  edit(id: number, employee: Employee): Observable<Employee> {
    return this.httpClient.put<Employee>(`${API_URL}/employee/${id}`, employee);
  }

  delete(id: number) {
    return this.httpClient.delete(`${API_URL}/employee/${id}`);
  }
}
